cardano-client/src/Cardano/Client/API.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Client.API
```

Comment:
I think it's worth to control what we export from this module. Otherwise it will be too easy to make changes to this module and not realize that they need to be propagated.

Then make sure that this module exports everything that the functions exported here needs..

Reply:
Also a better name for this module would be `Cardano.Client.Subscription`

Reply:
For now I only export the `subscribe` function
I want to figure out the exact exports later after this is tested in the applications.
This is just the first iteration.

Reply:
If you'd use `cardano-haskell` you could play with other repos before merging it.